src/object/schedcontext.c

```diff
@@ -105,6 +108,13 @@ static exception_t decodeSchedContext_Bind(sched_context_t *sc, extra_caps_t ext
             return EXCEPTION_SYSCALL_ERROR;
         }
 
+        if (isBlocked(TCB_PTR(cap_thread_cap_get_capTCBPtr(cap))) && (!sc_active(sc) || !refill_sufficient(sc, 0)
```

Comment:
In the API I tried very hard to avoid creating functions where the user would have to have total knowledge of what the system is doing to be able to alter the SC, specifically for dynamically tuning timing or handling faults.

Reply:
This only prevents binding an SC to TCB that is faulted if that SC isn't ready (which would mean that it was taken from another TCB that was not able to run due to lack of time for some reason). It passes all the tests, although it may rule out cases where you are managing SCs in more esoteric ways that involve moving SCs between TCBs.